src/ddmd/root/filename.d

```diff
@@ -721,3 +731,132 @@ nothrow:
         return str;
     }
 }
+
+version(Windows)
+{
+    /*
+      The code before used the POSIX function `mkdir` on Windows. That
+      function is now deprecated and fails with long paths, so instead
+      we use the newer `CreateDirectoryW`.
+
+      `CreateDirectoryW` is the unicode version of the generic macro
+      `CreateDirectory`.  `CreateDirectoryA` has a file path
+      limitation of 248 characters, `mkdir` fails with less and might
+      fail due to the number of consecutive `..`s in the
+      path. `CreateDirectoryW` also normally has a 248 character
+      limit, unless the path is absolute and starts with `\\?\`. Note
+      that this is different from starting with the almost identical
+      `\\?`.
+
+      Please consult
+      https://msdn.microsoft.com/en-us/library/windows/desktop/aa363855(v=vs.85).aspx
+    */
+    private int _mkdir(const(char)* path) nothrow
+    {
+        import core.stdc.errno: errno, EEXIST, ENOENT;
+
+        // see core.sys.windows.winerror - the reason it's not imported here is because
+        // the autotester's dmd is too old and doesn't have that module
+        enum NO_ERROR = 0;
+        enum ERROR_PATH_NOT_FOUND = 3;
+        enum ERROR_ALREADY_EXISTS = 183;
+
+        SetLastError(NO_ERROR);
+        const createRet = path.extendedPathThen!(p => CreateDirectoryW(&p[0],
+                                                                       null /*securityAttributes*/));
+        const lastError = GetLastError();
+
+        // Preserve compatibility with mkdir since the calling code expects
+        // errno to be set.
```

Comment:
Don't need compatibility since there is only one caller of this private function. Please don't use `errno` on Windows unless forced to.

Reply:
Fixed.